### ISSUE TITLE
Implement TTIRToTTNN pass to capture a TTNN.SoftmaxOp

### ISSUE THREAD

**Issue by mrakitaTT** [tenstorrent/tt-mlir#689]
StableHLO doesn't have a Softmax OP and implements it as a combination of simpler OPs. We need to detect this pattern of converted TTIR ops and match it with TTNN.SoftmaxOp.

**Comment by staylorTT** [comment]
@mrakitaTT Did we make a call on what to do with this?

**Comment by mrakitaTT** [comment]
This was blocked on implementing fusion pass in tt-mlir, which recently happened so we can now attend to this task. This is a perf improvement so I am not sure about priority, I've now raised attention to this and couple other similar tasks [here](https://github.com/tenstorrent/tt-mlir/issues/688#issuecomment-2891570568).

**Comment by mrakitaTT** [comment]
Seems like somebody has already implemented the fusion pass for softmax [here](https://github.com/tenstorrent/tt-mlir/blob/bf7d8fc3a45f47f5ee2fe0803b40cfe03baa6a3e/lib/Dialect/TTIR/Transforms/TTIRFusing.cpp#L151), we need to check if it works for softmax from both JAX and PyTorch. We also want to cover it with composite method from frontend.
